src/blueapi/service/interface.py

```diff
@@ -75,131 +102,93 @@ def _publish_event_streams(streams_to_destinations: Mapping[EventStream, str]) -
     """Submit a task to be run on begin_task"""
-    _ensure_worker_started()
-    return _Singleton.worker.submit_task(task)
+    return worker().submit_task(task)
 
 
 def clear_task(task_id: str) -> str:
     """Remove a task from the worker"""
-    _ensure_worker_started()
-    return _Singleton.worker.clear_task(task_id)
+    return worker().clear_task(task_id)
 
 
 def begin_task(task: WorkerTask) -> WorkerTask:
     """Trigger a task. Will fail if the worker is busy"""
-    _ensure_worker_started()
     if task.task_id is not None:
-        _Singleton.worker.begin_task(task.task_id)
+        worker().begin_task(task.task_id)
     return task
 
 
 def get_tasks_by_status(status: TaskStatusEnum) -> list[TrackableTask]:
     """Retrieve a list of tasks based on their status."""
-    _ensure_worker_started()
-    return _Singleton.worker.get_tasks_by_status(status)
+    return worker().get_tasks_by_status(status)
 
 
 def get_active_task() -> TrackableTask | None:
     """Task the worker is currently running"""
-    _ensure_worker_started()
-    return _Singleton.worker.get_active_task()
+    return worker().get_active_task()
 
 
 def get_worker_state() -> WorkerState:
     """State of the worker"""
-    _ensure_worker_started()
-    return _Singleton.worker.state
+    return worker().state
 
 
 def pause_worker(defer: bool | None) -> None:
     """Command the worker to pause"""
-    _ensure_worker_started()
-    _Singleton.worker.pause(defer)
+    worker().pause(defer)
 
 
 def resume_worker() -> None:
     """Command the worker to resume"""
-    _ensure_worker_started()
-    _Singleton.worker.resume()
+    worker().resume()
 
 
 def cancel_active_task(failure: bool, reason: str | None) -> str:
     """Remove the currently active task from the worker if there is one
     Returns the task_id of the active task"""
-    _ensure_worker_started()
-    return _Singleton.worker.cancel_active_task(failure, reason)
+    return worker().cancel_active_task(failure, reason)
 
 
 def get_tasks() -> list[TrackableTask]:
     """Return a list of all tasks on the worker,
     any one of which can be triggered with begin_task"""
-    _ensure_worker_started()
-    return _Singleton.worker.get_tasks()
+    return worker().get_tasks()
 
 
 def get_task_by_id(task_id: str) -> TrackableTask | None:
     """Returns a task matching the task ID supplied,
     if the worker knows of it"""
-    _ensure_worker_started()
-    return _Singleton.worker.get_task_by_id(task_id)
-
-
-def get_state() -> bool:
-    """Initialization state"""
-    return _Singleton.initialized
+    return worker().get_task_by_id(task_id)
 
 
-def _ensure_worker_started() -> None:
-    if _Singleton.initialized:
-        return
-    raise Exception("Worker must be stared before it is used")
+def health_check() -> Mapping[str, bool]:
+    """Health of various components, returns a mapping of
+    component name to whether it is up and running"""
+    return ...
```

Comment:
Seems to be unused, is this for a seperate PR?

Reply:
I started going down the route of getting the main process to query the subprocess' health, but then, yes, decided that should be a separate issue.